packages/@storefront/flux-capacitor/test/unit/core/actions/validators.ts

```diff
@@ -652,4 +652,111 @@ suite('validators', ({ expect, spy, stub }) => {
       expect(validators.isNotFetching.func(forward, state)).to.be.true;
     });
   });
+
+  describe('hasValidLabels', () => {
+    it('should be valid if labels is not present', () => {
+      expect(validators.hasValidLabels.func(<any>{})).to.be.true;
+    });
+
+    it('should be valid if labels is an array of strings', () => {
+      expect(validators.hasValidLabels.func(<any>{ labels: ['a', 'b'] })).to.be.true;
+    });
+
+    it('should be invalid if labels is an empty array', () => {
+      expect(validators.hasValidLabels.func(<any>{ labels: [] })).to.be.false;
+    });
+
+    it('should be invalid if labels is an array of anything other than strings', () => {
+      expect(validators.hasValidLabels.func(<any>{ labels: [{ foo: 'baz' }, 'bar'] })).to.be.false;
+    });
+  });
+
+  describe('hasValidOptions', () => {
+    it('should be valid if options is an array of objects with the required fields', () => {
+      const payload: any = {
+        options: [
+          { field: 'foo' },
+          { field: 'bar', descending: false },
+          { field: 'baz', descending: true },
+        ],
+      };
+
+      expect(validators.hasValidOptions.func(payload)).to.be.true;
+    });
+
+    [
+      'foo',
+      1,
+      true,
+      {},
+      () => null,
+      undefined,
+    ].forEach((val) => {
+      it(`should be invalid if options is a ${typeof val}`, () => {
```

Comment:
If you say "is of type" instead of "is a", you won't find yourself in the awkward situation where you get "is a undefined". Ditto for the same below.

Reply:
Good point, will update.